src/Marten.Testing/Schema/UniqueIndexMultiTenantTests.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using Marten.Schema;
+using Marten.Storage;
+using Npgsql;
+using Shouldly;
+using Xunit;
+
+namespace Marten.Testing.Schema
+{
+
+    public class UniqueIndexMultiTenantTests
+    {
+        public const string UniqueSqlState = "23505";
+
+
+        private class Project
+        {
+            public Project()
+            {
+                Id = Guid.NewGuid();
+            }
+
+            public Guid Id { get; set; }
+
+            public string Name { get; set; }
+        }
+
+        private class Project2 : Project { } //used for duplicatedfield index tests
+
+        //used for attributes index tests
+        private class Project3 : Project
```

Comment:
Could you change name of this class from `Project3` to more descriptive - eg. `ProjectWithUniqueCodePerTenant`